tests/fast-integration/kcp/client.js

```diff
@@ -4,41 +4,32 @@ const {
   getEnvOrThrow,
   debug,
   wait,
+  isEnvSet,
 } = require('../utils');
 const {inspect} = require('util');
 
 class KCPConfig {
   static fromEnv() {
-    return new KCPConfig(
-        getEnvOrThrow('KCP_KEB_API_URL'),
-        getEnvOrThrow('KCP_OIDC_ISSUER_URL'),
-        getEnvOrThrow('KCP_GARDENER_NAMESPACE'),
-        getEnvOrThrow('KCP_TECH_USER_LOGIN'),
-        getEnvOrThrow('KCP_TECH_USER_PASSWORD'),
-        getEnvOrThrow('KCP_OIDC_CLIENT_ID'),
-        getEnvOrThrow('KCP_OIDC_CLIENT_SECRET'),
-        getEnvOrThrow('KCP_MOTHERSHIP_API_URL'),
-        getEnvOrThrow('KCP_KUBECONFIG_API_URL'),
-    );
+    return new KCPConfig();
   }
-  constructor(host,
-      issuerURL,
-      gardenerNamespace,
-      username,
-      password,
-      clientID,
-      clientSecret,
-      motherShipApiUrl,
-      kubeConfigApiUrl) {
-    this.host = host;
-    this.issuerURL = issuerURL;
-    this.gardenerNamespace = gardenerNamespace;
-    this.username = username;
-    this.password = password;
-    this.clientID = clientID;
-    this.clientSecret = clientSecret;
-    this.motherShipApiUrl = motherShipApiUrl;
-    this.kubeConfigApiUrl = kubeConfigApiUrl;
+  constructor() {
+    this.host = getEnvOrThrow('KCP_KEB_API_URL');
+    this.issuerURL = getEnvOrThrow('KCP_OIDC_ISSUER_URL');
+    this.gardenerNamespace = getEnvOrThrow('KCP_GARDENER_NAMESPACE');
+    this.username =  getEnvOrThrow('KCP_TECH_USER_LOGIN');
+    this.password = getEnvOrThrow('KCP_TECH_USER_PASSWORD');
+    this.clientID = getEnvOrThrow('KCP_OIDC_CLIENT_ID');
+
+    if (isEnvSet('KCP_OIDC_CLIENT_SECRET')) {
+      this.clientSecret = getEnvOrThrow('KCP_OIDC_CLIENT_SECRET');
+    } else {
+      this.oauthClientID = getEnvOrThrow('KCP_OAUTH2_CLIENT_ID');
+      this.oauthSecret = getEnvOrThrow('KCP_OAUTH2_CLIENT_SECRET');
+      this.oauthIssuer = getEnvOrThrow('KCP_OAUTH2_ISSUER_URL');
+    }
+
+    this.motherShipApiUrl = getEnvOrThrow('KCP_MOTHERSHIP_API_URL');
+    this.kubeConfigApiUrl = getEnvOrThrow('KCP_KUBECONFIG_API_URL');
```

Comment:
looks like linter is unhappy about this section
https://status.build.kyma-project.io/view/gs/kyma-prow-logs/pr-logs/pull/kyma-project_kyma/16173/pre-main-kyma-integration-k3d/1595731542741618688
```
/home/sa_117798148653314453801/kyma/tests/fast-integration/kcp/client.js
   19:20  error  Multiple spaces found before 'getEnvOrThrow'                 no-multi-spaces
  116:17  error  Block must not be padded by blank lines                      padded-blocks
  118:5   error  Unexpected var, use let or const instead                     no-var
  118:13  error  Missing semicolon                                            semi
  119:5   error  Expected space(s) after "if"                                 keyword-spacing
  122:24  error  Trailing spaces not allowed                                  no-trailing-spaces
  332:15  error  'getReconciliationsInfo' is assigned a value but never used  no-unused-vars
```